aurora-postgresql-plugin/src/test/java/co/cask/auroradb/postgres/AuroraPostgresSourceTestRun.java

```diff
@@ -0,0 +1,325 @@
+    Assert.assertEquals(125.45, (double) row2.get("NUMERIC_COL"), 0.000001);
+    Assert.assertEquals(124.45, (double) row1.get("DECIMAL_COL"), 0.000001);
+    // Verify time columns
+    java.util.Date date = new java.util.Date(CURRENT_TS);
+    SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd");
+    LocalDate expectedDate = Date.valueOf(sdf.format(date)).toLocalDate();
+    sdf = new SimpleDateFormat("H:mm:ss");
+    LocalTime expectedTime = Time.valueOf(sdf.format(date)).toLocalTime();
+    ZonedDateTime expectedTs = date.toInstant().atZone(ZoneId.ofOffset("UTC", ZoneOffset.UTC));
+    Assert.assertEquals(expectedDate, row1.getDate("DATE_COL"));
+    Assert.assertEquals(expectedTime, row1.getTime("TIME_COL"));
+    Assert.assertEquals(expectedTs, row1.getTimestamp("TIMESTAMP_COL", ZoneId.ofOffset("UTC", ZoneOffset.UTC)));
+
+    // verify binary columns
+    Assert.assertEquals("user1", Bytes.toString(((ByteBuffer) row1.get("BYTEA_COL")).array(), 0, 5));
+    Assert.assertEquals("user2", Bytes.toString(((ByteBuffer) row2.get("BYTEA_COL")).array(), 0, 5));
+  }
+
+  @Test
+  public void testDbSourceMultipleTables() throws Exception {
+    String importQuery = "SELECT \"my_table\".\"ID\", \"your_table\".\"NAME\" FROM \"my_table\", \"your_table\"" +
+      "WHERE \"my_table\".\"ID\" < 3 and \"my_table\".\"ID\" = \"your_table\".\"ID\" and $CONDITIONS";
+    String boundingQuery = "SELECT MIN(MIN(\"my_table\".\"ID\"), MIN(\"your_table\".\"ID\")), " +
+      "MAX(MAX(\"my_table\".\"ID\"), MAX(\"your_table\".\"ID\"))";
+    String splitBy = "\"my_table\".\"ID\"";
+    ETLPlugin sourceConfig = new ETLPlugin(
+      AuroraPostgresConstants.PLUGIN_NAME,
+      BatchSource.PLUGIN_TYPE,
+      ImmutableMap.<String, String>builder()
+        .putAll(BASE_PROPS)
+        .put(AbstractDBSource.DBSourceConfig.IMPORT_QUERY, importQuery)
+        .put(AbstractDBSource.DBSourceConfig.BOUNDING_QUERY, boundingQuery)
+        .put(AbstractDBSource.DBSourceConfig.SPLIT_BY, splitBy)
+        .put(Constants.Reference.REFERENCE_NAME, "DBMultipleTest")
+        .build(),
+      null
+    );
+
+    String outputDatasetName = "output-multitabletest";
+    ETLPlugin sinkConfig = MockSink.getPlugin(outputDatasetName);
+
+    ApplicationManager appManager = deployETL(sourceConfig, sinkConfig,
+                                              DATAPIPELINE_ARTIFACT, "testDBSourceWithMultipleTables");
+    runETLOnce(appManager);
+
+    // records should be written
+    DataSetManager<Table> outputManager = getDataset(outputDatasetName);
+    List<StructuredRecord> outputRecords = MockSink.readOutput(outputManager);
+    Assert.assertEquals(2, outputRecords.size());
+    String userid = outputRecords.get(0).get("NAME");
+    StructuredRecord row1 = "user1".equals(userid) ? outputRecords.get(0) : outputRecords.get(1);
+    StructuredRecord row2 = "user1".equals(userid) ? outputRecords.get(1) : outputRecords.get(0);
```

Comment:
the rows come in a nondeterministic order?

Reply:
These lines are also legacy from old plugin test. From database perspective the order is deterministic (i.e. we read rows in order we wrote them) but I'm not sure about test pipeline implementation, for example it might simulate data shuffling, etc.